[Archives of the Empire: Vol 2.] When creating a character, the Greased Goat Star Sign fails to add Animal Affinity

To replicate: create a character using character creation, while making sure to  choose the Greased Goat. Finish character creation. The following errors appear in the console and the Animal Affinity talent is not added to the character sheet.

foundry.js:58721 Error when running effect Animal Affinity, please see the console (F12)
fetch @ foundry.js:58721

wfrp4e.js:5476 Error when running effect Animal Affinity - If this effect comes from an official module, try replacing the actor/item from the one in the compendium. If it still throws this error, please use the Bug Reporter and paste the details below, as well as selecting which module and 'Effect Report' as the label.
runSingleEffect @ wfrp4e.js:5476
wfrp4e.js:5477 REPORT
-------------------
EFFECT:	Animal Affinity
ACTOR:	Henrich Burgund - h6GvC8gAhWwqYne0
ERROR:	SyntaxError: await is only valid in async functions and the top level bodies of modules
runSingleEffect @ wfrp4e.js:5477
<br/>**From**: The First#7330<br/>foundry: 10.288<br/>wfrp4e: 6.2.1<br/>wfrp4e-archives1: 2.4.1<br/>wfrp4e-archives2: 3.0.0<br/>wfrp4e-core: 3.0.2<br/>wfrp4e-dotr: 2.3.4<br/>wfrp4e-eis: 2.3.4<br/>wfrp4e-empire-ruins: 3.0.1<br/>wfrp4e-horned-rat: 2.0.1<br/>wfrp4e-pbtt: 2.2.3<br/>wfrp4e-rnhd: 3.0.0<br/>wfrp4e-starter-set: 3.0.0<br/>wfrp4e-ua1: 2.3.2<br/>Active Modules: about-face, colorsettings, compendium-folders, dd-import, drag-ruler, easy-target, fast-flip, foundryvtt-simple-calendar, lib-wrapper, mcc, popout, quick-insert, settings-extender, smalltime, socketlib, token-action-hud, tokenmagic, wfrp4e-gm-toolkit, forge-vtt, forge-vtt-optional

--- a/modules/actor/actor-wfrp4e.js
+++ b/modules/actor/actor-wfrp4e.js
@@ -374,10 +374,11 @@ export default class ActorWfrp4e extends Actor {
         size = "avg";
     }
 
-    this.runEffects("calculateSize", {size})
+    let args = {size}
+    this.runEffects("calculateSize", args)
 
     // If the size has been changed since the last known value, update the value 
-    this.details.size.value = size || "avg"
+    this.details.size.value = args.size || "avg"
 
     if (this.flags.autoCalcSize && game.actors) {
       let tokenData = this._getTokenSize();

--- a/modules/actor/actor-wfrp4e.js
+++ b/modules/actor/actor-wfrp4e.js
@@ -2700,11 +2700,11 @@ export default class ActorWfrp4e extends Actor {
       }
     }
 
-    let engagedEffect = weapon.parent.conditions.find(x => x.id == "engaged");
+    let engagedEffect = weapon.parent.conditions.find(x => x.statusId == "engaged");
     if (engagedEffect) { 
-      modifier = 0;
+      modifier = Math.min(0, weapon.range.bands[currentBand]?.modifier || 0);
       tooltip.push(`${game.i18n.localize("EFFECT.ShooterEngaged")}`);
-    } 
+    }
     else {
       modifier += weapon.range.bands[currentBand]?.modifier || 0;
       if (modifier) {

--- a/modules/actor/sheet/actor-sheet.js
+++ b/modules/actor/sheet/actor-sheet.js
@@ -518,13 +518,13 @@ export default class ActorSheetWfrp4e extends ActorSheet {
                    do {
                     let testObject = duplicate(test);
                     let testDuplicate = test.constructor.recreate(testObject.data);
-                    await testDuplicate.roll();
                     if (testDuplicate.result.minormis || testDuplicate.result.majormis || testDuplicate.result.catastrophicmis) {
                       break;
                     }
                     if (test.item.cn.SL >= test.item.cn.value) {
                       break;
                     }
+                    await testDuplicate.roll();
                   } while (true);
                 } else {
                   await test.roll();

--- a/modules/hooks/entryContext.js
+++ b/modules/hooks/entryContext.js
@@ -137,7 +137,7 @@ export default function () {
 
           if (li.find(".dice-roll").length) {
             let amount = li.find('.dice-total').text();
-            game.user.targets.forEach(t => t.actor.applyBasicDamage(amount))
+            canvas.tokens.controlled.map(i => i.document.actor).concat(Array.from(game.user.targets).map(i => i.document.actor)).forEach(a => a.applyBasicDamage(amount))
           }
           else {
             let message = game.messages.get(li.attr("data-message-id"))
@@ -158,7 +158,7 @@ export default function () {
         callback: li => {
           if (li.find(".dice-roll").length) {
             let amount = li.find('.dice-total').text();
-            game.user.targets.forEach(t => t.actor.applyBasicDamage(amount, { damageType: game.wfrp4e.config.DAMAGE_TYPE.IGNORE_AP }))
+            canvas.tokens.controlled.map(i => i.document.actor).concat(Array.from(game.user.targets).map(i => i.document.actor)).forEach(a => a.applyBasicDamage(amount, { damageType: game.wfrp4e.config.DAMAGE_TYPE.IGNORE_AP }))
           }
           else {
             let message = game.messages.get(li.attr("data-message-id"))
@@ -179,7 +179,7 @@ export default function () {
         callback: li => {
           if (li.find(".dice-roll").length) {
             let amount = li.find('.dice-total').text();
-            game.user.targets.forEach(t => t.actor.applyBasicDamage(amount, { damageType: game.wfrp4e.config.DAMAGE_TYPE.IGNORE_TB }))
+            canvas.tokens.controlled.map(i => i.document.actor).concat(Array.from(game.user.targets).map(i => i.document.actor)).forEach(a => a.applyBasicDamage(amount, { damageType: game.wfrp4e.config.DAMAGE_TYPE.IGNORE_TB }))
           }
           else {
             let message = game.messages.get(li.attr("data-message-id"))
@@ -200,7 +200,7 @@ export default function () {
         callback: li => {
           if (li.find(".dice-roll").length) {
             let amount = li.find('.dice-total').text();
-            game.user.targets.forEach(t => t.actor.applyBasicDamage(amount, { damageType: game.wfrp4e.config.DAMAGE_TYPE.IGNORE_ALL }))
+            canvas.tokens.controlled.map(i => i.document.actor).concat(Array.from(game.user.targets).map(i => i.document.actor)).forEach(a => a.applyBasicDamage(amount, { damageType: game.wfrp4e.config.DAMAGE_TYPE.IGNORE_ALL }))
           }
           else {
             let message = game.messages.get(li.attr("data-message-id"))

--- a/modules/system/opposed-test.js
+++ b/modules/system/opposed-test.js
@@ -1,4 +1,5 @@
 import WFRP_Audio from "./audio-wfrp4e.js";
+import WomCastTest from "./rolls/wom-cast-test.js";
 import WFRP_Utility from "./utility-wfrp4e.js";
 
 
@@ -320,7 +321,7 @@ export default class OpposedTest {
     //@/HOUSE
 
     // Winds of Magic overcast
-    if (item.type == "spell" && game.settings.get("wfrp4e", "useWoMOvercast")) {	
+    if (this.attackerTest instanceof WomCastTest) {	
       damage += (this.attackerTest.result.additionalDamage || 0);	
     } else {	
       damage += (opposedSL + (this.attackerTest.result.additionalDamage || 0));	

--- a/modules/system/rolls/wom-cast-test.js
+++ b/modules/system/rolls/wom-cast-test.js
@@ -42,7 +42,8 @@ export default class WomCastTest extends CastTest {
       if (this.item.Damage && this.result.castOutcome == "success") {
         this.result.damage = Number(this.item.Damage)
         if (this.result.overcast.usage.damage && this.result.overcast.usage.damage.count > 0) {
-          this.result.damage += game.wfrp4e.config.overCastTable.damage[this.result.overcast.usage.damage.count - 1].value
+          this.result.additionalDamage += game.wfrp4e.config.overCastTable.damage[this.result.overcast.usage.damage.count - 1].value
+          this.result.damage += this.result.additionalDamage
         }
       }
       if (this.item.damage.dice && !this.result.additionalDamage) {

--- a/modules/system/socket-handlers.js
+++ b/modules/system/socket-handlers.js
@@ -39,7 +39,7 @@ export default class SocketHandlers  {
         let actor = new ActorWfrp4e(data.payload.actorData)
         let effect = new EffectWfrp4e(data.payload.effect)
         
-        game.wfrp4e.utility.runSingleEffect(effect, actor, null, {actor});
+        game.wfrp4e.utility.runSingleEffect(effect, actor, null, {actor}, { async: true});
     }
     static changeGroupAdvantage(data){
         if (!game.user.isGM || !game.settings.get("wfrp4e", "useGroupAdvantage")) 

--- a/modules/system/utility-wfrp4e.js
+++ b/modules/system/utility-wfrp4e.js
@@ -1093,7 +1093,7 @@ export default class WFRP_Utility {
       }
     }
 
-    WFRP_Utility.runSingleEffect(effect, actor, null, { actor });
+    WFRP_Utility.runSingleEffect(effect, actor, null, { actor}, {async: true});
   }
 
   static async runSingleEffect(effect, actor, item, scriptArgs, options = {}) {

--- a/static/templates/apps/chargen/career.html
+++ b/static/templates/apps/chargen/career.html
@@ -46,7 +46,7 @@
     {{/if}}
 
     {{#if (gte context.step 2)}}
-    <p class="centered">Keep rolling (no extra experience)</p>
+    <p class="centered">{{localize "CHARGEN.Career.KeepRolling"}}</p>
     <div class="buttons">
       <a class='chargen-button' data-button='rollCareers'>{{localize "Roll"}}</a>
     </div>

--- a/static/templates/apps/chargen/skills-talents.html
+++ b/static/templates/apps/chargen/skills-talents.html
@@ -19,10 +19,10 @@
 
       <div class="flexcol">
 
-        <div class="drag-area" data-advance="5">
+        <div class="drag-area" data-advance="3">
           <p class="centered">{{localize "CHARGEN.SkillsTalents.NumAdvances" num=3}}</p>
           <div class="flexrow">
-            {{#each speciesSkillAllocation.[5] as |skill index|}}
+            {{#each speciesSkillAllocation.[3] as |skill index|}}
             <a class="drag-skill item-lookup" data-type="skill">
               {{skill}}
             </a>
@@ -30,10 +30,10 @@
           </div>
         </div>
 
-        <div class="drag-area" data-advance="3">
+        <div class="drag-area" data-advance="5">
           <p class="centered">{{localize "CHARGEN.SkillsTalents.NumAdvances" num=5}}</p>
           <div class="flexrow">
-            {{#each speciesSkillAllocation.[3] as |skill index|}}
+            {{#each speciesSkillAllocation.[5] as |skill index|}}
             <a class="drag-skill item-lookup" data-type="skill">
               {{skill}}
             </a>

--- a/static/templates/apps/chargen/trappings.html
+++ b/static/templates/apps/chargen/trappings.html
@@ -46,7 +46,7 @@
     <div>
       {{#each context.added as |trapping index|}}
       <a class="content-link" draggable="true" data-uuid="{{trapping.uuid}}"><i class="fas fa-suitcase"></i>{{trapping.name}}</a>
-      <a class="remove-trapping" data-index="{{index}}"><i class="fa-solid fa-x"></i></a>
+      <a class="remove-trapping" data-index="{{index}}"><i class="fa-solid fa-times"></i></a>
       {{/each}}
     </div>
 

